content/5_Plat/Range_Sweep.mdx

```diff
@@ -139,42 +179,65 @@ template <class T> struct Seg {  // comb(ID,b) = b
 	}
 };
 
+const int MX = 1e5;
 Seg<int> S;
 int N, P;
 int ans[MX];
 vi distinct_y;
 
 int query_min(int ind) { return S.query(0, ind); }
 void ins(int ind, int val) { S.upd(ind, val); }
-int y_index(int y) { return lb(all(distinct_y), y) - begin(distinct_y); }
+// gets the coordinate compressed y
+int y_index(int y) {
+	return lower_bound(all(distinct_y), y) - begin(distinct_y);
+}
 
 int main() {
-	setIO("boards");
+	ios_base::sync_with_stdio(0);
+	cin.tie(0);
+	freopen("boards.in", "r", stdin);
+	freopen("boards.out", "w", stdout);
 	cin >> N >> P;
-	vector<pair<pair<int, int>, pair<int, int>>> ev;
+	vector<pair<pair<int, int>, pair<int, int>>> events;
 	for (int i = 0; i < P; ++i) {
 		pair<int, int> a, b;
 		cin >> a.f >> a.s >> b.f >> b.s;
-		ev.push_back({a, {i, -1}});  // start point
-		ev.push_back({b, {i, 1}});   // end point
+		events.pb({a, {i, -1}});  // start point
+		events.pb({b, {i, 1}});   // end point
 		distinct_y.pb(a.s);
 		distinct_y.pb(b.s);
 	}
+
 	sort(all(distinct_y));
-	sort(begin(ev), end(ev));
+	sort(all(events));
 	S.init(2 * P);
 	ins(0, 0);
-	for (auto &t : ev) {
+	for (auto &t : events) {
 		if (t.s.s == -1) {
 			ans[t.s.f] = t.f.f + t.f.s + query_min(y_index(t.f.s));
 		} else {
 			ins(y_index(t.f.s), ans[t.s.f] - t.f.f - t.f.s);
 		}
 	}
-	cout << query_min(2 * P - 1) + 2 * N;
+	cout << query_min(2 * P - 1) +
+	            2 * N;  // adds 2N to include the transition from the last
+	                    // springboard to the endpoint
 }
 ```
 
+Bonus: Think of this problem as maximizing the distance covered by springboards to simplify the solution.
```

Comment:
what is this supposed to mean

Reply:
The answer is 2N - [distance covered by springboards]. If thought of in this way, you do not need to do the ans[j] - x2[j] - y2[j] thing in the segtree.

Reply:
Do you want me to remove this

Reply:
no i just dk what this means lol